### ISSUE TITLE
Add a [skip travis] command

### ISSUE THREAD

**Issue by alexislefebvre** [travis-ci/travis-ci#5032]
When updating my repository in order to update tests executed by Windows only, I would like to launch all the CI builds except Travis, but I can't tell only to Travis to ignore this commit.

I think that a commit containing `[skip travis]` (or [travis`skip]`, see #911) should let users disable builds only for Travis CI.

It's possible to disable some builds on other websites:
- Scrutinizer: use [`[skip Scrutinizer]`](https://scrutinizer-ci.com/docs/guides/skipping_a_build_via_commit_message)
- AppVeyor: use [`[skip appveyor]`](https://github.com/travis-ci/travis-ci/issues/3005#issuecomment-71320541) (but it must be configured)

**Comment by JeremybellEU** [comment]
`[ci skip]` [does this for you](http://docs.travis-ci.com/user/customizing-the-build/#Skipping-a-build)

**Comment by alexislefebvre** [comment]
@JeremybellEU: I've edited my issue, it was unclear. If I add `[ci skip]` to a commit all the CI tools will ignore this commit, I would like to tell only to Travis to ignore a commit.

**Comment by JeremybellEU** [comment]
Yes that seems totally reasonable. :+1:

**Comment by skaller** [comment]
Agree. I use both Travis and Appveyor. [ci skip] works now with Appveyor. Would be good for Travis to also skip. [appveyor skip] and [travis skip] would make sense for both for selective skipping.

Actually even further control could be interesting: [travis set <JSON here>] where the JSON overrides or merges with the control file JSON .travis (implied by the YAML).

**Comment by alexislefebvre** [comment]
@skaller: can you please post another issue for your second paragraph? It's a different need than skipping Travis only.

**Comment by skaller** [comment]
@alexislefebvre : Its an extension of the idea. Imagine the .travis.yml file had a field skip: false and [travis {skip:true}] would just override the .travis.yml value for THAT commit.

With that interpretation, why not override other variables? For example, if you have matrix of builds, you might want to select only some of them. There is YAML to do that already in .travis.yml if you want. So a possible scenario is to set up a build matrix, but then blacklist all the builds except one. Since you just want to test the basics work on each commit. Occasionally you want to remove the blacklist to enable all builds. So in this case it is really the *same* idea :-) I'm not sure its worthy of a feature request though.

**Comment by bashtage** [comment]
This would be a welcome addition when trying to get issues like Windows-specific problems fixed.

**Comment by tharen** [comment]
Adding [skip travis] would be great to avoid conflicts with other CI platforms.

**Comment by alexislefebvre** [comment]
It should be possible to stop tests by adding a bash condition at the beginning of the Travis CI build, by parsing the last commit message and returning 0 if `[skip travis]` is found. However this would still launch builds, that's not a perfect solution.

**Comment by geowurster** [comment]
As mentioned in [another comment](https://github.com/travis-ci/travis-ci/issues/5032#issuecomment-172033144), this would be incredibly useful for debugging Windows issues in Appveyor.

**Comment by cyberium** [comment]
No interest on this? Should be easy/not risky enough to be added fastly...

**Comment by montefra** [comment]
+1 from me

**Comment by maartenbreddels** [comment]
indeed, when debugging on appveyor I do a lot of unneeded builds on travis, kind of a waste of resources.

**Comment by alexislefebvre** [comment]
I created a one-liner that skip Travis builds of `[skip travis]` is found in the last commit message:

```bash
if git show -s HEAD | grep -F -q "[skip travis]" ; then echo "skip travis" && exit 0 ; fi
```

As an example, here is a gist with this line in a complete `.travis.yml` file: https://gist.github.com/alexislefebvre/597ad49e5f60e804ea2d45b73de9967d

If you have some suggestions, please add comments on this gist instead of commenting on this issue.

**Comment by ConAntonakos** [comment]
What if you want to process the build, but skip testing. Is that possible?

**Comment by alexislefebvre** [comment]
@ConAntonakos Please explain more precisely what you do with Travis CI. You can skip the tests if you `return 0` between the build and the tests.

**Comment by ConAntonakos** [comment]
@alexislefebvre Thanks for the response. Sorry if I wasn't clear. I meant I make a minor change, I trigger the build for deployment, but I don't want to run my unit tests all over again. Is that possible?

**Comment by alexislefebvre** [comment]
We can now use the commit message provided by Travis CI: https://github.com/travis-ci/travis-build/pull/936

**Comment by mloskot** [comment]
Cheers for linking the https://github.com/htgoebel/pyinstaller/commit/a19a8855953ee27b855e9a6abaf9aa402bdd3358 to this issue. Helps greatly to find the neat solution ready.

The trivial `[skip travis]` could have been already implemented dozens of times, and people requesting it here would have wasted less electricity...

**Comment by maxkoryukov** [comment]
How to vote on this?

**Comment by aantron** [comment]
This would be very useful for me. It already exists on AppVeyor, why not add it to Travis? When I'm debugging an AppVeyor build, or setting one up, I don't want a ton of notifications from errored Travis builds. There are other situations as well.

**Comment by alexislefebvre** [comment]
Please stop adding comments to say only _me too_ (it sends email and GitHub notifications for nothing). If you want to vote, put a _thumbs up_ emoji on the first message instead. Thanks.

**Comment by solvingj** [comment]
Note: [skip appveyor] no longer needs to be configured, it's supported by default.

**Comment by SlySven** [comment]
Whilst I would wish to avoid appearing as a mere _me too_ comment, I would remain _da Management_ that the Public MacOS is operating now with a pretty much continuous back-log.  As a coder for a project that uses CI for all three of Linux, MacOS and Windows (with AppVeyor) we really want to help you by not dropping unneeded builds into the system.  Right now I am trying to debug an AppVeyor issue but it is painful to see the extra jobs piling up on Travis and even though I try to manually stop the builds by keeping the `https://travis-ci.org/Project/Project/pull_requests` page open I am concerned they still enter the system.

tl;dr; Getting this feature in (by checking for `[skip travis]` AS WELL AS `[skip ci]` OR `[ci skip]` and not creating a build in that case either) could help to reduce the load on your public platform at least and enable your users to help you. 🙏

**Comment by jakirkham** [comment]
@BanzaiMan, please correct me if I'm wrong. Does this just come down to changing [this line]( https://github.com/travis-ci/travis-core/blob/fd905d451087627e1d7320585b10c5bf75d6ad40/lib/travis/commit_command.rb#L9 ) and adding a test [here]( https://github.com/travis-ci/travis-core/blob/fd905d451087627e1d7320585b10c5bf75d6ad40/spec/travis/commit_command_spec.rb )? If so, are PRs from the community welcome?

**Comment by SlySven** [comment]
Ironically it is the backwards skip that it makes the most sense to clobber as `[skip ci]` should be extended to `[skip ci]` or `[skip travis]` in a case insensitive way.  OTOH `[ci travis]` does not make sense to prevent _normal_ behaviour like `[ci skip]` does - that is more like a command to enable a CI run for Travis for _this_ commit/PR/what-have-you when it is not configured to run automatically (which is a whole _separate_ issue I believe! 😀 )

**Comment by SlySven** [comment]
@jakirkham how about putting up that PR - it seems you will need to target the correct sub-division rather than here as this central issue tracking repository is not the place to issue PRs to - but a cross-reference would be helpful to all parties watching this topic. 😀

**Comment by Lekensteyn** [comment]
Same issue here, I'm pushing changes just to test with AppVeyor, but Travis is needlessly wasting resources building it. Workaround: push all changes to an `appveyor` branch and add this to `.travis.yml`:
```
branches:
  except:
    - appveyor
```
Regexes can also be used (e.g. `- /appveyor\/.*/` will skip every branch starting with `appveyor/`).
https://docs.travis-ci.com/user/customizing-the-build/#building-specific-branches

**Comment by meatballhat** [comment]
This is now implemented, FYI: https://changelog.travis-ci.com/skip-travis-ci-builds-with-skip-travis-in-commit-message-74620
